apps/data-pipeline/map-scraper/src/index.ts

```diff
@@ -0,0 +1,126 @@
+
+const LOCATIONS_LIST_API =
+  "https://api.concept3d.com/locations?map=463&key=0001085cc708b9cef47080f064612ca5";
+const LOCATIONS_DETAIL_API = (id: number) =>
+  `https://api.concept3d.com/locations/${id}?map=463&key=0001085cc708b9cef47080f064612ca5`;
+const locationsCatalogue: Record<number, BuildingLocation> = {};
+const locationIds: Record<string, number> = {};
+
+function sleep(ms: number): Promise<number> {
+  return new Promise((resolve) => setTimeout(() => resolve(ms), ms));
+}
+
+async function fetchLocations() {
+  const locationsListResponse: Response = await fetch(LOCATIONS_LIST_API);
+  const locations: LocationListData[] = (await locationsListResponse.json()) as LocationListData[];
+  for (const location of locations) {
+    // Check if location was already recorded
+    const locationExists: boolean = Object.values(locationsCatalogue).some(
+      (existingLocation) => existingLocation.name === location.name,
+    );
+
+    if (!CATEGORIES.has(location.catId)) {
+      continue;
+    }
+    if (locationExists) {
+      continue;
+    }
+
+    // If one of the locations belongs to one of the categories above, and is not a duplicate
+    // Hits location data API to get details on location
+    const locationsDetailResponse: Response = await fetch(LOCATIONS_DETAIL_API(location.id));
+    const locationData: LocationDetailData =
+      (await locationsDetailResponse.json()) as LocationDetailData;
+
+    const imgUrls: string[] = [];
+
+    // Collects image URLs by checking if type is an image
+    // { 'mediaUrlTypes': ['image', '...', 'image'], 'mediaUrls': ['xyz.jpg', '...', 'abc.png']}
+    if (locationData.mediaUrlTypes !== undefined) {
+      for (const [i, media] of locationData.mediaUrlTypes.entries()) {
+        if (media === "image") {
+          imgUrls.push(locationData.mediaUrls[i]);
+        }
+      }
+    }
+
+    locationsCatalogue[location.id] = {
+      name: location.name,
+      lat: location.lat,
+      lng: location.lng,
+      imageURLs: imgUrls,
+    };
+
+    const locationName: string = location.name.includes("(")
+      ? location.name.substring(location.name.indexOf("(") + 1, location.name.indexOf(")"))
+      : location.name;
+    locationIds[locationName] = location.id;
+  }
+}
+
+async function main() {
+  const url = process.env.DB_URL;
+  if (!url) throw new Error("DB_URL not found");
+  const db = database(url);
+  console.log("map scraper starting...");
+  await fetchLocations();
+  console.log("Done fetching locations...");
+  await db.transaction(async (tx) => {
+    for (const locationId in locationsCatalogue) {
+      const locationData = locationsCatalogue[locationId];
+      await tx
+        .insert(mapLocation)
+        .values({
+          id: locationId,
+          name: locationData.name,
+          latitude: locationData.lat.toString(),
+          longitude: locationData.lng.toString(),
+          imageURLs: locationData.imageURLs,
+        })
+        .onConflictDoUpdate({
```

Comment:
why not update every column on conflict? is there a good reason to retain the `imageURLs` column on conflict?